examples/denoise_covar2d.py

```diff
@@ -77,23 +79,21 @@
 imgs_clean = vol2img(vols[..., 0], rots)
 
 # Assign the CTF information and index for each image
-h_idx = sim.filters.indices
-filters = sim.filters.filters
+h_idx = np.array([filters.index(f) for f in sim.filters])
```

Comment:
There's no need for any *internal* code within `aspire` to keep track of indices. Not sure why this example script needs it. My suspicion is that whatever `h_idx` is being used for, it can be done away with. Something to bring up with Junchao next week.

Reply:
For cov2d denoising, the `h_idx` is needed to track the ctf filter for each imag because the mean and covariance values are computed by the ctf groups.  In experimental dataset, the `h_idx` information can be read in as metadata of images.

Reply:
Couldn't the CTF groups be extracted by working with the `filters` array instead of these proxy indices?

Reply:
In Vineet latest change, the `filters.index(f)` gets the index of filter `f`. It might be more reasonable to include an indices as a metadata in the `Image` object if we decide to include all metadata in the `Image` class. In the real experimental dataset, the indices to the CTF filters should be built when read in the star file from the CTF finding process.

Reply:
I'm closing this here. An upcoming `IndexedTransformer` or something similar will allow the user to attaching one or more `Transformer` objects in a `Pipeline` attached to an `ImageSource`, and we'll add a method there to return the index (or indices) of any `Transformer` that the caller provides.